### PR TITLE
Firehose Logging Part 1

### PR DESCRIPTION
**Pull request by hannahbergam**
This is Part 1 of the deprecate Firehose eng excellence work! I wanted this to be a small move to give folks a feel for what this work might look like.

This PR only removes firehose events where there is a corresponding Amplitude event in place. It covers files in both the TT and Acquisitions world, so tagging both teams for a quick look.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/sectionProgress/SectionProgress.jsx [hunk 1]
```diff
@@ -89,11 +84,6 @@ class SectionProgress extends Component {
...
     this.props.setLessonOfInterest(lessonOfInterest);
 
-    this.recordEvent('jump_to_lesson', {
-      script_id: this.props.scriptId,
-      stage_id: this.props.scriptData.lessons[lessonOfInterest].id,
-    });
-
     analyticsReporter.sendEvent(EVENTS.PROGRESS_JUMP_TO_LESSON, {
       sectionId: this.props.sectionId,
       unitId: this.props.scriptId,
```
Comment (TurnerRiley) [inline apps/src/templates/sectionProgress/SectionProgress.jsx]: I think line 105 and the `recordEvent` function can be removed here too?

File: apps/src/code-studio/pd/application/teacher/TeacherApplication.jsx [hunk 2]
```diff
@@ -33,19 +32,8 @@ const autoComputedFields = [
...
-      user_id: userId,
-      study: 'application-funnel',
-      event: event,
-    },
-    {includeUserId: false}
-  );
-};
-
 const TeacherApplication = props => {
-  const {savedFormData, accountEmail, userId, savedStatus, schoolId} = props;
```
Comment (TurnerRiley) [inline apps/src/code-studio/pd/application/teacher/TeacherApplication.jsx:48]: If we no longer need the `userId` prop here, do we need it in the `new` method of the `teacher_application_controller.rb`?

**Review by TurnerRiley — APPROVED** [review]
Yes thank you so much for kicking this off! Just a couple questions but overall looks great!

**Comment by dju90** [comment]
With the removal of events from the progress tab, I think my main concern comes from possibly losing access to up-to-date data/learnings on this Tableau dashboard: https://us-east-1.online.tableau.com/#/site/codeorg/workbooks/613727/views, which I still use occasionally - interactions with the progress view feeds into some of the reports, such as tab switching and progress tab activity.
In general, for the firehose removal project, there are some really rich reports in https://us-east-1.online.tableau.com/#/site/codeorg/projects/173624 that don't currently have an equivalent in Amplitude, and I'm wondering if we can make a plan on how to preserve the value we get out of them (e.g. can we create equivalent events + dashboards in Amplitude first?) before we remove the firehose events.

**Comment by dju90** [comment]
I could also see a world in which Product decides that the numbers in those Tableau dashboards is unlikely to change, and so we're okay with the snapshot that we have today. But basically, I would urge us to check in with the Product team to make sure we're aligned with whatever loss of access to data/learnings would be the consequence of the removal.

File: apps/src/code-studio/pd/application/teacher/TeacherApplication.jsx (hunk 2 above)
Comment (hannahbergam) [reply apps/src/code-studio/pd/application/teacher/TeacherApplication.jsx:48]: Good catch- thank you! Removed!

File: apps/src/templates/sectionProgress/SectionProgress.jsx (hunk 1 above)
Comment (hannahbergam) [reply apps/src/templates/sectionProgress/SectionProgress.jsx]: Ideally yes I will in the future! The only issue here is that there's not yet a corresponding Amplitude event for that line:/

**Comment by davidsbailey** [comment]
hi @hannahbergam , where can I learn more about the plan for this effort? What are we planning to use for logging in cases where amplitude events do not already exist?

**Comment by hannahbergam** [comment]
> hi @hannahbergam , where can I learn more about the plan for this effort? What are we planning to use for logging in cases where amplitude events do not already exist?

Hi @davidsbailey, here is a brain dump draft of a plan I made this morning! I am currently only working through duplicate instances, and was going to tackle the more sticky challenges of non-Amplitude-represented firehose logging in the future, but not without checking in with product first and creating new Amplitude events as necessary. https://docs.google.com/document/d/1AwjauChxXFILBgz_tI1PhZO5gRA69RIX_O4Apv0ZtFs/edit

**Comment by davidsbailey** [comment]
thanks @hannahbergam , makes sense. I agree with the change you've got here, however I'd encourage you to start working on a proposal for moving the remaining firehose events to amplitude which addresses the following:
* does amplitude meet the needs of developers? specifically, can amplitude by queried via SQL or another similarly precise form of analysis?
* does amplitude have enough capacity to move over all firehose events? if not, can we get under capacity by eliminating a few of the most frequent firehose events?

**Comment by hannahbergam** [comment]
> With the removal of events from the progress tab, I think my main concern comes from possibly losing access to up-to-date data/learnings on this Tableau dashboard: https://us-east-1.online.tableau.com/#/site/codeorg/workbooks/613727/views, which I still use occasionally - interactions with the progress view feeds into some of the reports, such as tab switching and progress tab activity. In general, for the firehose removal project, there are some really rich reports in https://us-east-1.online.tableau.com/#/site/codeorg/projects/173624 that don't currently have an equivalent in Amplitude, and I'm wondering if we can make a plan on how to preserve the value we get out of them (e.g. can we create equivalent events + dashboards in Amplitude first?) before we remove the firehose events.

I've removed all firehose deprecation from progress files, leaving this only to be a Teacher Application update for now.

**Pull request merged**